index.html

```diff
@@ -3036,11 +2995,34 @@ <h3><dfn>Navigate To</dfn></h3>
   with <var>session</var> and <a>current top-level browsing
   context</a>.
 
- <li><p>If <var>session</var>&apos;s <a>current top-level browsing context</a> contains
+ <!-- This redirect handling is extremely badly specified -->
+
+ <li><p>While <var>session</var>&apos;s <a>current top-level browsing context</a> contains
```

Comment:
Well the problem is the entire bit as currently phrased is nonsense. I started this trying to work out where _timer_ is coming from in "If a _timer_ has not been started […]", and the answer seems to be that we expect some of the variables to survive "Return to the first step of this algorithm". But apparently we don't expect that we'll restart the navigation to the URL in _command parameters_ otherwise we'd (probably) end up in an infinite loop.

I don't agree with the reading that the current spec only handles a single redirect; if gecko does that seems like a bug. If we restart the algorithm then (ignoring the reasons above that this doesn't work) presumably we get back to this step again, and if there's a second redirect we again restart, and so on. My change is just turning the (very) broken recursion into (still, but slightly less) broken iteration.

Reply:
I took the chance and checked what we do and stumbled over https://bugzilla.mozilla.org/show_bug.cgi?id=1363418. So we actually don't support this kind of redirect at all. But the work I'm currently doing to use the ProgressListener might help us to maybe get this feature added.

Given that Alex is fine as well here, lets close the issue.